bloom-housing/bloom
# feat: initial connect of public site to prisma

# Pull Request Template

## Issue Overview

This PR addresses #3615 


## Description

This PR does the following initial steps to connect the public site to the new Prisma based back-end:
- Moves the swagger file to the shared-helpers
  - This allows both front-ends to have access to this file and know all that is needed about the backend.
  - Down the road we might want to put this and other files in a separate package that is published to NPM, but for now this is all we need
- Connect the listings page to the backend. Or in other words all of the data when going to /listings is retrieved from the Prisma backend and appropriate changes have been made.
- Connect the listing details page to the backend

Notes:
- OrderBy functionality on the listings endpoint was not working because the name of the orderBy does not sync one to one with the field in question. That has been updated as part of this PR
- One bug around the limit field was also fixed on the backend.

Questions for @YazeedLoonat:
1. ~~countyCode appears to have not been added to listing. This is used in several places on the front-end. Want to know if this was intentional or not.~~ This was implemented by using Jurisdictions.name instead
2. ~~OrderBy doesn't work with the simplified logic (combined for multiple controllers). I've pulled back in the logic for listings. I wonder if this solution works, or were you picturing something else.~~ confirmed with Yazeed that this is the best solution for now. Going forward we should make the frontend use the same name on the fields as we use on the backend
3. ~~ListingEvents - ?~~ Discovered bug in the DTO. This has been resolved
4. ~~Is there a better way to select all but a few fields in a query. See listing service line 60 for reference~~ Updated the listingFeature and ListingUtilities DTOs to not be extensions of the AbstractDTO

## How Can This Be Tested/Reviewed?

- Add the new backend endpoint to the public site .env file `BACKEND_API_BASE_NEW=http://localhost:3101`
- Update the jurisdiction name in public site .env file to the same one that is seeded `JURISDICTION_NAME=Bloomington`
- Reseed the prisma backend
  1. Go to backend_new directory
  5. run `yarn db:setup:staging`
- Start up the prisma backend
- Start up the public site
- Go to http://localhost:3000/listings and see the listings

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
